pkg/prometheus/promcfg.go

```diff
@@ -2238,15 +2247,18 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	return cfg
 }
 
-func (cg *ConfigGenerator) generateTracingConfig(tracingConfig *monitoringv1.PrometheusTracingConfig) (yaml.MapItem, error) {
+func (cg *ConfigGenerator) generateTracingConfig() (yaml.MapItem, error) {
 	cfg := yaml.MapSlice{}
 	objMeta := cg.prom.GetObjectMeta()
 
-	cfg = append(cfg, yaml.MapItem{
-		Key:   "endpoint",
-		Value: tracingConfig.Endpoint,
-	})
+	tracingConfig := cg.prom.GetCommonPrometheusFields().TracingConfig
 
+	if len(tracingConfig.Endpoint) > 0 {
```

Comment:
API validation should already ensure that this isn't an empty string

Reply:
Yes, it is fine :)

Reply:
@ArthurSens done